python/lsst/ap/association/association.py

```diff
@@ -114,30 +114,39 @@ def run(self,
               not matched a new DiaSource. (`int`)
         """
         diaSources = self.check_dia_source_radec(diaSources)
+
+        if self.config.doTrailedSourceFilter:
+            diaTrailedResult = self.trailedSourceFilter.run(diaSources,
+                                                            exposure_time)
+            diaSources = diaTrailedResult.diaSources
+            trailedSources = diaTrailedResult.trailedDiaSources
+
+            self.log.info(
+                "%i DIASources exceed max_trail_length, dropping fromsource catalog." % len(
+                    diaTrailedResult.trailedDiaSources))
+            self.metadata.add("num_filtered",
+                              len(diaTrailedResult.trailedDiaSources))
+        else:
+            trailedSources = pd.DataFrame(columns=diaSources.columns)
```

Comment:
Consider a comment along the lines of, "Create an empty trailedSources df to add to the return Struct." It's a little counterintuitive to me that "trailedSources" is only populated if we decide to drop long trailed sources. Makes me wonder what happens to any short trailed sources...

Reply:
The intention of trailed sources is just to contain the sources that will be dropped, however we may want to think of renaming it?

Reply:
I think I'm partial to `longTrailedSources` throughout, if that's not too much of a headache. (If they go off the edge, odds are good they are long, so no need to distinguish between edge overlapping trailed sources vs long trailed sources that may or may not intersect an edge IMO)